solidity/contracts/isms/hook/AbstractHookISMV3.sol

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity >=0.8.0;
+
+/*@@@@@@@       @@@@@@@@@
+ @@@@@@@@@       @@@@@@@@@
+  @@@@@@@@@       @@@@@@@@@
+   @@@@@@@@@       @@@@@@@@@
+    @@@@@@@@@@@@@@@@@@@@@@@@@
+     @@@@@  HYPERLANE  @@@@@@@
+    @@@@@@@@@@@@@@@@@@@@@@@@@
+   @@@@@@@@@       @@@@@@@@@
+  @@@@@@@@@       @@@@@@@@@
+ @@@@@@@@@       @@@@@@@@@
+@@@@@@@@@       @@@@@@@@*/
+
+// ============ Internal Imports ============
+
+import {IInterchainSecurityModule} from "../../interfaces/IInterchainSecurityModule.sol";
+import {Message} from "../../libs/Message.sol";
+import {TypeCasts} from "../../libs/TypeCasts.sol";
+
+// ============ External Imports ============
+
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {Initializable} from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+
+/**
+ * @title ArbtractNativeISM
+ * @notice Uses the native bridges to verify interchain messages.
+ * @dev In the future, the hook might be moved inside the Mailbox which doesn't require storage mappings for senders.
+ *      for more details see https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/2381
+ * @dev V3 WIP
+ */
+abstract contract AbstractHookISM is IInterchainSecurityModule, Initializable {
```

Comment:
I might just call this something like
`MessageIDAuthIsm` or something
verifying message ID from a specific sender

Reply:
also imo we can move the `verifyMessageId` thing into here
and just have the auth code be bridge specific

Reply:
done